xtevent/_eventols.ado

```diff
@@ -159,17 +170,98 @@ program define _eventols, rclass
 		}
 	}
 	else loc indepvars ""
+		
+	******** SA *******
+	if "`sun_abraham'"!=""{
+		* Parse the dependent variable
+		local lhs = "`depenvar'"
+		local rel_time_list = "`included'"
+		
+		* Convert the varlist of relative time indicators to nvarlist 
+		local nvarlist "" // copies of relative time indicators with control cohort set to zero
+		local dvarlist "" // for display
+		foreach l of varlist `rel_time_list' {
+			local dvarlist "`dvarlist' `l'"
+			tempname n`l'
+			qui gen `n`l'' = `l'
+			qui replace `n`l'' = 0 if  `control_cohort' == 1 
+			local nvarlist "`nvarlist' `n`l''" 
+		}	
+
+		* Get cohort count  and count of relative time
+		qui levelsof `cohort' if  `control_cohort' == 0, local(cohort_list) 
+		local nrel_times: word count `nvarlist' 
+		local ncohort: word count `cohort_list'  
+		
+		**** step 2 
+		* Initiate empty matrix for weights 
+		tempname bb ff_w
+		
+		* Loop over cohort and get cohort shares for relative times
+		local nresidlist ""
+		foreach yy of local cohort_list {
+			tempvar cohort_ind resid`yy'
+			qui gen `cohort_ind'  = (`cohort' == `yy') 
+			qui regress `cohort_ind' `nvarlist'  if `touse' & `control_cohort' == 0 [`weight'`exp']  , nocons
+			mat `bb' = e(b) 
+			matrix `ff_w'  = nullmat(`ff_w') \ `bb' 
+			*di "`yy'"
+			*mat li `ff_w'
+			qui predict double `resid`yy'', resid 
+			local nresidlist "`nresidlist' `resid`yy''" //list of variables of residuals 
+		}
+		
+		* Get VCV estimate for the cohort shares using avar
+		* In case users have not set relative time indicators to zero for control cohort
+		* Manually restrict the sample to non-control cohort
+		tempname XX Sxx Sxxi S KSxxi Sigma_ff
+		qui mat accum `XX' = `nvarlist' if  `touse' & `control_cohort' == 0 [`weight'`exp'], nocons
+		mat `Sxx' = `XX'*1/r(N) 
+		mat `Sxxi' = syminv(`Sxx') 
+		qui avar (`nresidlist') (`nvarlist')  if `touse' & `control_cohort' == 0 [`weight'`exp'], nocons robust
+		mat `S' = r(S) 
+		mat `KSxxi' = I(`ncohort')#`Sxxi' 
+		mat `Sigma_ff' = `KSxxi'*`S'*`KSxxi'*1/r(N) 
+		// Note that the normalization is slightly different from the paper
+		// The scaling factor is 1/N for N the obs of cross-sectional units
+		// But here estimates are on the panel, which is why it is 1/NT instead
+		// Should cancel out for balanced panel, but unbalanced panel is a TODO
+		

```

Comment:
@Constantino-Carreto-Romero What should be the proper scaling here? Can you please check SA's code again to see if they changed this?

Reply:
@Constantino-Carreto-Romero thanks. Please add this info about the scaling factor in page 50 and the last code update date as a comment here, to make sure we don't forget when we revisit this.

Reply:
In https://github.com/JMSLab/xtevent/commit/a15a73e40baf57bd02e669253733a4458b9216d5 and https://github.com/JMSLab/xtevent/commit/e28ed455b3ff8f10fdf34466500eb1d7111eff56 I implemented.